proxy/src/serverless/sql_over_http.rs

```diff
@@ -244,11 +244,44 @@ pub(crate) async fn handle(
     backend: Arc<PoolingBackend>,
     cancel: CancellationToken,
 ) -> Result<Response<BoxBody<Bytes, hyper::Error>>, ApiError> {
+    let backend_clone = Arc::clone(&backend);
     let result = handle_inner(cancel, config, &ctx, request, backend).await;
 
     let mut response = match result {
         Ok(r) => {
             ctx.set_success();
+
+            // Handling the error response from local proxy here
+            if let crate::auth::Backend::ControlPlane(..) = backend_clone.auth_backend {
```

Comment:
This code path is not used in regular proxies?

Reply:
Hm, actually I found out that it might affect ws/http path so I changed the condition :)